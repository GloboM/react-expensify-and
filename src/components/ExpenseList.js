import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
import { startSetExpenses, startAddExpense } from '../actions/expenses';

export class ExpenseList extends React.Component {
  
  render () {
    return (
      <div>
        {
          this.props.expenses.length === 0 ? (
            <p>No expenses</p>
          ) : (
              this.props.expenses.map((expense) => {
                return <ExpenseListItem key={expense.id} {...expense} />;
              })
            )
        }
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters), 
  };
};


export default connect(mapStateToProps)(ExpenseList);
