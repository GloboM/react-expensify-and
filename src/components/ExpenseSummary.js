
import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import expensesSelect from '../selectors/expenses';
import expensesTotalSelect from '../selectors/expensesTotal';

export const ExpenseSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const expensesTotalFormatted = numeral(expensesTotal).format('$0,0.00');
    return(
        <div>
        <h1>You are Viewing {expensesCount} {expenseWord} totalling {expensesTotalFormatted}</h1>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = expensesSelect(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: expensesTotalSelect(visibleExpenses)
    }

}
export default connect(mapStateToProps)(ExpenseSummary);