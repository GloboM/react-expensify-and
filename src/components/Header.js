
import React from 'react';
import { connect } from 'react-redux';

import { startSignOut } from '../actions/authentication';
import { NavLink } from 'react-router-dom';

export const Header = ({startSignOut}) => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/dashboard" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    <button onClick={startSignOut}>Log out</button>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startSignOut: () => dispatch(startSignOut())
})
export default connect(undefined, mapDispatchToProps)(Header);
