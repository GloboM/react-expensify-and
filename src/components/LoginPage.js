
import React from 'react';
import { connect } from 'react-redux';
import { startAuthentication } from '../actions/authentication';

export const LoginPage = (props) => {
    const onLogin = () => {
        props.startAuthentication();
        //props.history.push("/dashboard");
    }
    return (
        <div>
            <button onClick={onLogin}>Login</button>
        </div>
    )
};

const mapDispatchToProps = (dispatch) => ({
    startAuthentication : () => dispatch(startAuthentication())
})
export default connect(null, mapDispatchToProps)(LoginPage);