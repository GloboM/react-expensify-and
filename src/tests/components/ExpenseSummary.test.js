
import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test ('should render expenses summary correctly with one expense', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={1} expenseTotal={200}/>);
    expect(wrapper).toMatchSnapshot();
})

test('should render expenses summary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={3} expensesTotal={450123}/>);
    expect(wrapper).toMatchSnapshot();
})