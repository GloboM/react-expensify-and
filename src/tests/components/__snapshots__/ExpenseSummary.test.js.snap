// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary correctly with multiple expenses 1`] = `
<div>
  <h1>
    You are Viewing 
    3
     
    expenses
     totalling 
    $450,123.00
  </h1>
</div>
`;

exports[`should render expenses summary correctly with one expense 1`] = `
<div>
  <h1>
    You are Viewing 
    1
     
    expense
     totalling 
    $0.00
  </h1>
</div>
`;
